app/components/add-cron-job.js

```diff
@@ -1,45 +1,76 @@
 import Component from '@ember/component';
 import { inject as service } from '@ember/service';
-import { task } from 'ember-concurrency';
+import { task, timeout } from 'ember-concurrency';
+import { mapBy } from '@ember/object/computed';
+import Branch from 'travis/models/branch';
+import config from 'travis/config/environment';
 
 export default Component.extend({
   store: service(),
 
   classNames: ['form--cron'],
 
+  intervals: ['Monthly', 'Weekly', 'Daily'],
+
+  options: [{
+    name: 'Always run',
+    value: false
+  }, {
+    name: 'Do not run if there has been a build in the last 24h',
+    value: true
+  }],
+
+  currentCronJobsBranches: mapBy('repository.cronJobs', 'branch.name'),
+
+  init() {
+    this.reset();
+    this._super(...arguments);
+  },
+
   reset() {
     this.setProperties({
       selectedBranch: null,
-      selectedInterval: null,
-      disable: null
+      selectedInterval: this.intervals.firstObject,
+      selectedOption: this.options[0]
     });
   },
 
-  save: task(function* () {
-    const store = this.get('store');
-    const repoId = this.get('branches.firstObject.repoId');
-    const branch = this.get('selectedBranch') || this.get('branches.firstObject');
-
-    const existingCrons = yield store.filter('cron', { repository_id: repoId }, (c) => {
-      c.get('branch.repoId') === repoId && c.get('branch.name') === branch.get('name');
+  search: task(function* (query) {
+    yield timeout(config.intervals.searchDebounceRate);
+    let branchNames = this.get('currentCronJobsBranches');
+    let branches = yield this.store.query('branch', {
+      repository_id: this.repository.id,
+      data: {
+        name: query,
+        sort_by: 'name',
+        limit: 10,
+        exists_on_github: true
+      }
```

Comment:
As I mentioned before, it's not necessary to wrap params with `data`.

Reply:
Please take a look here: https://github.com/travis-ci/travis-web/blob/2adf33c31687303b6770724f7d1e65a86e3c59ba/app/components/add-cron-job.js#L47-L51

And it's deployed here: https://as-crons-improvements.test-deployments.travis-ci.org/travis-ci/travis-web/settings

So you can see live how it works.